cmd/kubeadm/app/cmd/phases/upgrade/node/kubeletconfig.go

```diff
@@ -71,8 +71,18 @@ func runKubeletConfigPhase() func(c workflow.RunData) error {
 			return err
 		}
 
-		// TODO: Checkpoint the current configuration first so that if something goes wrong it can be recovered
-
+		// Create a copy of the kubelet config file in the /etc/kubernetes/tmp/ folder.
+		backupDir, err := constants.CreateTempDirForKubeadm(constants.KubernetesDir, "kubeadm-kubelet-config")
+		if err != nil {
+			return err
+		}
+		src := filepath.Join(kubeletDir, constants.KubeletConfigurationFileName)
+		dest := filepath.Join(backupDir, constants.KubeletConfigurationFileName)
+		fmt.Printf("[upgrade] backing up kubelet config file to %s\n", dest)
+		if err := os.Rename(src, dest); err != nil {
+			return errors.Wrap(err, "error backing up the kubelet config file")
+		}
+		fmt.Printf("[upgrade] kubelet config file is moved to %s\n", dest)
```

Comment:
```suggestion
```

its ok to remove this print.
point was- print before the action

Reply:
Done